openapi_spec_validator/validators.py

```diff
@@ -339,10 +341,10 @@ def __init__(self, dereferencer):
 
     @wraps_errors
     def iter_errors(self, schema, value):
-        validator = OAS30Validator(
+        validator = OAS31Validator(
```

Comment:
Shouldn't this be dependent of which spec we use? Otherwise we will use OAS31Validator for all specs? 🤔

Reply:
Good catch, I will look into this, create a test case and cover it in a dedicated PR.

Reply:
We don't really have any information about the spec at this step. I would sugest that we add The validator class and the format checker instance as two additional configuration parameters to the `SpecValidator`

```
openapi_v30_spec_validator = SpecValidator(
    openapi_v30_validator_factory,
    resolver_handlers=default_handlers,
    validator_class=OAS30Validator,
    format_checker=oas30_format_checker
)
```

This also means they need to be propagated on each new validator class. What are your thoughts about this or do you have any other suggestions?